### PR TITLE
ticdc/redo: should cleanup redo logs when removing a paused changefeed

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Owner should cleanup redo logs when removing a paused changefeed

### What is changed and how it works?

- Extract some ignoralbe errors for cdc processor
- Add a `redoManagerCleanup` function, it is called in when changefeed calls `releaseResources`, no matter whether the changefeed has been initialized.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
Fix a bug that redo logs are not cleaned up when removing a paused changefeed.
```

### CODE REVIEW

File: cdc/processor/processor.go [hunk 1]
```diff
@@ -109,6 +109,25 @@ func newProcessor(ctx cdcContext.Context) *processor {
 	return p
 }
 
+// processorIgnoralbeError returns true if the error means the processor exits
+// normally, caused by changefeed pause, remove, etc.
+func processorIgnoralbeError(err error) bool {
```
Comment (ben1009) [inline cdc/processor/processor.go:114]: may be could include err==nil case, since already extract as a func

**Review by ben1009 — APPROVED** [review]

File: cdc/processor/processor.go (hunk 1 above)
Comment (amyangfei) [reply cdc/processor/processor.go]: addressed

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**